### PR TITLE
Drop jsonschema

### PR DESCRIPTION
**Pull request by braingram**
PRs open for downstream users:
- https://github.com/DKISTDC/dkist/pull/295
- https://github.com/BAMWelDX/weldx/pull/886
- https://github.com/spacetelescope/jwst/pull/7880

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -18,6 +18,7 @@ The ASDF Standard is at v1.6.0
 - Convert numpy scalars to python types during yaml encoding
   to handle NEP51 changes for numpy 2.0 [#1605]
 - Vendorize jsonschema 4.17.3 [#1591]
+- Drorp jsonschema as a dependency [#1614]
```
[the remark this answers is not in the document]
Comment (braingram) [reply CHANGES.rst]: Speling is Hawrd :) Thanks! Fixed in: 97bcc6dbbfd298064e807bd6b259a8f513743698

File: asdf/exceptions.py [hunk 2]
```diff
@@ -10,6 +10,9 @@
 ]
 
 
+ValidationError.__module__ = "asdf.exceptions"
```
[the remark this answers is not in the document]
Comment (braingram) [reply asdf/exceptions.py]: Good catch. Overwriting the module does break `inspect.getsource` (which appears to rely on `getmodule`).

I added this to fix the doctest to make it not contain a check for `asdf._jsonschema.exceptions.ValidationError` and instead check for `asdf.exceptions.ValidationError` and as an attempt to keep '_jsonschema' hidden. I don't think it's worth the breaking of `inspect.getsource` so I undid the change and updated the doctest.

**Comment by braingram** [comment]
Remaining jwst failures are unrelated see: https://github.com/spacetelescope/jwst/pull/7874

**Comment by zacharyburnett** [comment]
would this also fix this issue in the tests: https://github.com/spacetelescope/stenv/actions/runs/6162954331/job/16725854445#step:8:76
```
E   AttributeError: 'Namespace' object has no attribute 'jsonschema'
```

**Comment by braingram** [comment]
> would this also fix this issue in the `stenv` tests: https://github.com/spacetelescope/stenv/actions/runs/6162954331/job/16725854445#step:8:76
> 
> ```
> E   AttributeError: 'Namespace' object has no attribute 'jsonschema'
> ```
> 
> EDIT: After talking with @braingram we're going to try running the `stenv` `asdf` tests from source and see if that fixes the failures in the short term [spacetelescope/stenv#116](https://github.com/spacetelescope/stenv/pull/116)

Thanks for following up about this. Doing some local testing if I run things with python 3.10 and pip the installed package tests run fine with `pytest --pyargs asdf` (installed from pypi). I'm only able to replicate the issue if I run things with conda.

**Pull request merged**